omp/matrix/csr_kernels.cpp

```diff
@@ -127,6 +129,199 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_CSR_ADVANCED_SPMV_KERNEL);
 
 
+template <typename ValueType, typename IndexType>
+void spgemm_insert_row(std::unordered_set<IndexType> &cols,
+                       const matrix::Csr<ValueType, IndexType> *c,
+                       size_type row)
+{
+    auto row_ptrs = c->get_const_row_ptrs();
+    auto col_idxs = c->get_const_col_idxs();
+    cols.insert(col_idxs + row_ptrs[row], col_idxs + row_ptrs[row + 1]);
+}
+
+
+template <typename ValueType, typename IndexType>
+void spgemm_insert_row2(std::unordered_set<IndexType> &cols,
+                        const matrix::Csr<ValueType, IndexType> *a,
+                        const matrix::Csr<ValueType, IndexType> *b,
+                        size_type row)
+{
+    auto a_row_ptrs = a->get_const_row_ptrs();
+    auto a_col_idxs = a->get_const_col_idxs();
+    auto b_row_ptrs = b->get_const_row_ptrs();
+    auto b_col_idxs = b->get_const_col_idxs();
+    for (size_type a_nz = a_row_ptrs[row];
+         a_nz < size_type(a_row_ptrs[row + 1]); ++a_nz) {
+        auto a_col = a_col_idxs[a_nz];
+        auto b_row = a_col;
```

Comment:
`auto b_row = a_col_idxs[a_nz];` should work. or do you use it for readability?

Reply:
Mostly for readability, so we have a conceptual separation between column indices and row indices. But that is a question of personal taste, so I have no issue with simplifying it.

Reply:
I am also fine with it.